services/traction/api/db/models/base.py

```diff
@@ -65,6 +77,29 @@ async def update_by_id(cls, item_id: UUID, values: dict):
                 raise e
 
 
+class TenantScopedModel(BaseModel):
+    tenant_id: uuid.UUID = Field(foreign_key="tenant.id", index=True)
+
+    @classmethod
+    def tenant_select(cls, fallback_tenant_id: UUID = None) -> Select:
+        from api.endpoints.dependencies.tenant_security import get_from_context
+
+        result = None
+        tenant_context_id = get_from_context("TENANT_ID")
+        # load from starlette context
+        if tenant_context_id:
+            result = select(cls).where(cls.tenant_id == tenant_context_id)
+        # manually passed in fallback for now
+        elif fallback_tenant_id:
+            result = select(cls).where(cls.tenant_id == fallback_tenant_id)
+        # couldn't load tenant context for some reason
+        else:
+            logger.error("not a http context, no tenant_id, EXECUTING AN UNSAFE QUERY")
```

Comment:
maybe make this a warning. and if the logger belongs to a class, we should be able to track it down